modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/endpointComponents/PathParameterConfiguration.tsx

```diff
@@ -0,0 +1,187 @@
+import {removeLeadingForwardSlash} from '../utils/string';
+
+interface PathParameterConfigurationProps {
+	data: Partial<APIEndpointUIData>;
+	selectedResponseBodySchema: SelectOption | undefined;
+	setData: Dispatch<SetStateAction<Partial<APIEndpointUIData>>>;
+}
+
+export default function PathParameterConfiguration({
+	data,
+	selectedResponseBodySchema,
+	setData,
+}: PathParameterConfigurationProps) {
+	const [selectedPathParameter, setSelectedPathParameter] = useState<
+		SelectOption
+	>();
+
+	const [pathParameterOptions, setPathParameterOptions] = useState<
+		SelectOption[]
+	>([]);
+
+	const handleSelectPathParameter = (value: string) => {
+		setData((previousValue) => ({
+			...previousValue,
+			pathParameter: value,
+		}));
+
+		setSelectedPathParameter(
+			pathParameterOptions.find((option) => option.value === value)
+		);
+	};
+
+	useEffect(() => {
+		setSelectedPathParameter(
+			pathParameterOptions.find(
+				(option) => option.value === data.pathParameter
+			)
+		);
+	}, [pathParameterOptions, data.pathParameter]);
+
+	const fetchSchemaByAPISchemaID = (
+		schemaId: string
+	): Promise<APISchemaItem> => {
+		return fetchJSON<APISchemaItem>({
+			input: `/o/headless-builder/schemas/${schemaId}`,
+		});
+	};
+
+	const fetchObjectDefinitionsByExternalReferenceCode = (
+		externalReferenceCode: string
+	): Promise<ObjectDefinition[]> => {
+		return getAllItems<ObjectDefinition>({
+			filter: `label eq 'ID' or label eq 'External Reference Code' or unique eq true`,
+			url: `/o/object-admin/v1.0/object-definitions/by-external-reference-code/${externalReferenceCode}/object-fields`,
+		});
+	};
+
+	const getObjectFieldLabel = (objectField: ObjectDefinition): string => {
+		if (
+			typeof objectField.label[
+				defaultLanguageId as keyof LocalizedValue<string>
+			] !== 'undefined'
+		) {
+			return objectField.label[
+				defaultLanguageId as keyof LocalizedValue<string>
+			]!;
+		}
+		else if (Object.keys(objectField.label).length !== 0) {
+			return Object.values(objectField.label)[0];
+		}
+		else {
+			return objectField.name;
+		}
+	};
+
+	const fetchPathParameterOptions = useCallback((schemaId: string) => {
+		fetchSchemaByAPISchemaID(schemaId).then((response) => {
+			fetchObjectDefinitionsByExternalReferenceCode(
+				response.mainObjectDefinitionERC
+			).then((response) => {
+				const options = response
+					? response.map((objectField) => ({
+							label: getObjectFieldLabel(objectField),
+							value: objectField.name,
+					  }))
+					: [];
+
+				if (options.length) {
+					setPathParameterOptions(options);
+				}
+			});
+		});
+	}, []);
+
+	useEffect(() => {
+		if (data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId) {
+			fetchPathParameterOptions(
+				data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId?.toString()
+			);
+		}
+	}, [
+		data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId,
+		fetchPathParameterOptions,
+	]);
+
+	return (
+		<>
+			{selectedResponseBodySchema ? (
```

Comment:
If it is not an if-else conditional, just use `selectedResponseBodySchema && (...)`
so you can remove this part
```
: (
  <></>
)
```